cli/commands/download.js

```diff
@@ -0,0 +1,193 @@
+const fs = require("fs");
+const path = require("path");
+const axios = require("axios");
+
+var ProgressBar = require("progress");
+
+const { hasApiKeyForWorkspace, getApiKeyForWorkspace } = require("../core.js");
+
+const api = require("../../api.js");
+
+function parseDatasetUrlWithVersion(datasetUrl) {
+    if (datasetUrl.includes("/")) {
+        const parts = datasetUrl.split("/");
+
+        if (parts.length == 3) {
+            return {
+                workspaceUrl: parts[0],
+                projectUrl: parts[1],
+                version: parts[2]
+            };
+        } else if (parts.length == 2) {
+            return {
+                workspaceUrl: parts[0],
+                projectUrl: parts[1],
+                version: null
+            };
+        } else {
+            throw new Error("invalid project url");
+        }
+    } else {
+        return {
+            workspaceUrl: null,
+            datasetUrl: datasetUrl,
+            version: null
+        };
+    }
+}
+
+async function getProjectVersions(workspaceUrl, projectUrl, apiKey) {
+    const project = await api.getProject(workspaceUrl, projectUrl, apiKey);
+
+    return project.versions || [];
+}
+
+async function getLatestVersion(workspaceUrl, projectUrl, apiKey) {
+    const availableVersions = await getProjectVersions(workspaceUrl, projectUrl, apiKey);
+    availableVersions.sort((a, b) => a.created - b.created);
+
+    if (availableVersions.length > 0) {
+        const versionUrl = availableVersions[availableVersions.length - 1].id;
+        const parsed = parseDatasetUrlWithVersion(versionUrl);
+
+        return parsed.version;
+    }
+
+    return null;
+}
+
+async function verifyVersionExists(workspaceUrl, projectUrl, version, apiKey) {
+    const availableVersions = await getProjectVersions(workspaceUrl, projectUrl, apiKey);
+    const versionExists = availableVersions.find((v) => v.id.endsWith(`/${version}`));
+    if (!versionExists) {
+        throw new Error(`version ${version} does not exist for dataset ${projectUrl}`);
```

Comment:
We actually will generate this automatically now! Cc @Jacobsolawetz for the details on that. I think it just auto-zips & returns back a progress amount if not completed so you can keep polling for it.

Reply:
Yeah it checks the progress and polls on the export already (for up to a minute)

This is for checking whether the version selected exists (e.g if you try to export/download version 10, but there is only 1,2,3 available)